### PR TITLE
Some fixes for BuildMappings and EnumerableDataReader

### PR DESCRIPTION
**Pull request by stiankroknes**
Fix for handling both ordinal and column name for destination mapping.
Fix for handling varbinary max and GetBytes impl.
Fix for handling nvarchar max.
Minor test improvements for BulkCopySetup.

### CODE REVIEW

**Comment by stiankroknes** [comment]
Fixes #19 and #15.

**Comment by jbogard** [comment]
This ready to review?

**Comment by stiankroknes** [comment]
Yes.

File: src/BulkWriter/Properties/Resources.resx [hunk 1]
```diff
@@ -140,4 +159,10 @@
   <data name="PropertyInfoExtensions_PropertyNotDeclaredOnType" xml:space="preserve">
     <value>The supplied property is not declared on a type.</value>
   </data>
+  <data name="EnumerableDataReader_GetBytes_OrdinalDoesNotMapToProperty" xml:space="preserve">
+    <value>The ordinal does not map to property</value>
```
Comment (jbogard) [inline src/BulkWriter/Properties/Resources.resx]: Could you add a period to these exception messages? See https://docs.microsoft.com/en-us/dotnet/standard/exceptions/best-practices-for-exceptions#use-grammatically-correct-error-messages

Comment (stiankroknes) [reply src/BulkWriter/Properties/Resources.resx]: Sure. I forgot this in my new PR.

**Pull request merged**